src/tracker-component.js

```diff
@@ -302,10 +302,12 @@ export const Tracker = (config? : Config = {}) => {
                 return;
             }
 
-            if (data.id) {
-                setUserId(data.id);
-            } else if (data.id === null) {
-                createNewUserId();
+            if (data.id || data.id === null) {
+                setMerchantProvidedUserId(data.id);
+
+                if (data.id === null) {
+                    setGeneratedUserId();
+                }
```

Comment:
could this logic be simpler here? 
```
if (data.id || data.id === 0) { // because 0 can be a valid id.
  setMerchantProvidedUserId(data.id)
} else { // I think else is better, it takes care of `undefined` cases. I don't think merchants will pass in null.
   setGeneratedUserId()
}
```

Reply:
`null` is specifically for unsetting a user, per the docs